removing as_any func

samuraikillers opened the pull request:
This PR resolve issue #576

badmonster0 commented:
Thanks for taking this! This PR fails to build - there're existing usages of `as_any()` not converted yet.

samuraikillers commented:
Hi @badmonster0 ,
So basically the change I made for the `ResourceSetupStatus` function changing static to Any, the same has to be done for others right?

badmonster0 commented:
> Hi @badmonster0 , So basically the change I made for the `ResourceSetupStatus` function changing static to Any, the same has to be done for others right?

@samuraikillers 
You removed `as_any()` but code using `as_any()` is still there, so the compile doesn't pass. Please see the check failures for more details: https://github.com/cocoindex-io/cocoindex/actions/runs/15361469976/job/43236701928?pr=579

You'll need to update the code using `as_any()`.

samuraikillers commented:
@badmonster0 that's exactly my question. The current push I did I have removed `as_any` inside the functions and added it in function type.

So I assume the same has to be done for all the code where we are observing this error?

badmonster0 commented:
> @badmonster0 that's exactly my question. The current push I did I have removed `as_any` inside the functions and added it in function type.
> 
> So I assume the same has to be done for all the code where we are observing this error?

@samuraikillers Of course, when a trait method is removed, all implementations for this trait method need to be cleaned up, and call sites for this method need to be migrated (for this specific case, using the new [trait upcasting](https://blog.rust-lang.org/2025/04/03/Rust-1.86.0/#trait-upcasting) feature added in Rust 1.86)

badmonster0 commented:
Closing this as this PR doesn't have much progress and no response for a week, and free up the original issue. Feel free open another PR if you addressed it.

vumichien commented:
@badmonster0 Can I work on this issue?
